upstairs/src/upstairs.rs

```diff
@@ -1756,11 +1759,38 @@ impl Upstairs {
                     Ok(NegotiationResult::WaitQuorum) => {
                         // Copy the region definition into the Downstairs
                         self.downstairs.set_ddef(self.ddef.get_def().unwrap());
-                        // See if we have a quorum
-                        if self.connect_region_set() {
-                            // We connected normally, so there's no need
-                            // to check for live-repair.
-                            self.repair_check_deadline = None;
+                        // If we are RO, then we have a different path than
+                        // if we are RW
+                        if self.cfg.read_only {
+                            if self.connect_ro_region_set() {
+                                // We are the first one to arrive, so we
+                                // can't be needing any live repair yet.
+                                self.repair_check_deadline = None;
+
+                                // XXX If we are trying to replace a region that
+                                // had never joined, do we need to do anything
+                                // different here with the repair_check_deadline?
```

Comment:
I have a Modest Proposal:

https://github.com/oxidecomputer/crucible/pull/1610

Reply:
Modest proposal approved.  I'll take this comment out and that will possibly refactor things a little if I no longer need to know first vs. second/third joining (remove the `-> bool` result).

Reply:
1610 has landed